### PR TITLE
test: add integration tests for the support api `AP-1527`

### PR DESCRIPTION
**Pull request by BryanttV**
## Description
This PR adds integration tests for the support API. These tests are executed in the job of **Tutor Integration Tests**

## Testing instructions
Check the jobs of **Tutor Integration Tests** in the PR.

## Jira Issue
- [AP-1527](https://edunext.atlassian.net/browse/AP-1527)

[AP-1527]: https://edunext.atlassian.net/browse/AP-1527?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .coveragerc
Comment (magajh) [inline .coveragerc:1]: Isn't this configuration already in https://github.com/eduNEXT/eox-core/blob/master/setup.cfg#L11-L21?

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (magajh) [inline eox_core/api/support/v1/tests/integration/test_views.py:71]: I know this PR is still a draft, so I’m holding off on a full review for now, but I think the formatting would look cleaner with:

```suggestion
class TestEdxAppUserAPIIntegration(
    SupportAPIRequestMixin,
    BaseIntegrationTest,
    UsersAPIRequestMixin,
):
```

File: .coveragerc
Comment (BryanttV) [reply .coveragerc:1]: Yes, you're right! I deleted the `.coveragerc` file: https://github.com/eduNEXT/eox-core/commit/10a050b96e4b3fb8095e05aba41292798e790170, ty

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (BryanttV) [reply eox_core/api/support/v1/tests/integration/test_views.py]: Done: https://github.com/eduNEXT/eox-core/commit/3ff25e8699849a907f7c6e184d18b8d7ee45e032

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (mariajgrimaldi) [inline eox_core/api/support/v1/tests/integration/test_views.py:125]: What if I try to delete a user from a different tenant? Should that work?

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (mariajgrimaldi) [inline eox_core/api/support/v1/tests/integration/test_views.py:167]: Same question here about updating a user from a different tenant. Should that work?

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (mariajgrimaldi) [inline eox_core/api/support/v1/tests/integration/test_views.py:253]: Could you explain in the docstring why creating the user is needed?

**Review by mariajgrimaldi — CHANGES_REQUESTED** [review]

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (BryanttV) [reply eox_core/api/support/v1/tests/integration/test_views.py:256]: The endpoint creates the user if the user doesn't exist. The conditional is to validate both cases. I updated the docstring: https://github.com/eduNEXT/eox-core/commit/bb37aab1015654d52b843dfae5ecc542dc8f9576

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (BryanttV) [reply eox_core/api/support/v1/tests/integration/test_views.py:125]: I added a test for this: https://github.com/eduNEXT/eox-core/commit/47e03c993866e0d63471354a2d0b1f2469e1aa71

File: eox_core/edxapp_wrapper/backends/users_m_v1.py
Comment (mariajgrimaldi) [reply eox_core/edxapp_wrapper/backends/users_m_v1.py:284]: But I don't think we should change this directly in Maple's backend, mainly because we're testing with quince, redwood, and nightly. Should we rename this backend to be compatible with the Redwood release instead?

**Comment by mariajgrimaldi** [comment]
LGTM, I have an additional comment for you to address: https://github.com/eduNEXT/eox-core/pull/300#discussion_r1836465656. Let me know once you do to leave my approval :)

File: eox_core/edxapp_wrapper/backends/users_m_v1.py
Comment (BryanttV) [reply eox_core/edxapp_wrapper/backends/users_m_v1.py:284]: Yes, we should rename the backend but, I tested in Quince, and the same thing happens. I'm not sure which version has the same behavior. How should we proceed in this case?

File: scripts/execute_tutor_extra_commands.sh
Comment (magajh) [inline scripts/execute_tutor_extra_commands.sh:2]: Would it be possible to just run this within the script that executes the test command and skip this script entirely?
[https://github.com/eduNEXT/integration-test-in-tutor/pull/8/files#r183854869](https://github.com/eduNEXT/integration-test-in-tutor/pull/8/files#r1838548698)

File: .github/workflows/integration-test.yml
Comment (magajh) [inline .github/workflows/integration-test.yml:10]: remember to undo this change before merging please

**Review by magajh — COMMENTED** [review]
Thank you for this @BryanttV! just have one comment about the extra commands script

**Review by magajh — DISMISSED** [review]
@BryanttV tag me if you need my approval again

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (BryanttV) [reply eox_core/api/support/v1/tests/integration/test_views.py:170]: I don't think it's necessary to check it here, anyway in those tests no users are being created.

File: eox_core/edxapp_wrapper/backends/users_m_v1.py
Comment (mariajgrimaldi) [reply eox_core/edxapp_wrapper/backends/users_m_v1.py:284]: Mmm. Then, since we still support quince, let's use q instead since it's the oldest version we support and we know that this backend works. What do you think?

Comment (BryanttV) [reply eox_core/edxapp_wrapper/backends/users_m_v1.py:284]: Yes, I'm agree. https://github.com/eduNEXT/eox-core/commit/cca23f4e2205ae8018044994859812f7f1cb8d5a

**Comment by BryanttV** [comment]
Hi @magajh @mariajgrimaldi, I made the last changes in the PR.

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (magajh) [reply eox_core/api/support/v1/tests/integration/test_views.py:170]: I understand that we aren’t creating any users in these tests, but is it possible that users created in previous tests might still exist? If I’m not mistaken, these tests don’t reset the database after each test case, so any user created would persist unless explicitly deleted. That’s why I think it could make sense to ensure our delete operation doesn’t unintentionally remove any users.

That said, this type of check might not be necessary, as it’s not directly testing an integration scenario. Still, it’s something we can think about further to define the boundaries of database state verification. For now, though, I think it’s fine as is

**Review by magajh — APPROVED** [review]

File: eox_core/api/support/v1/tests/integration/test_views.py
Comment (mariajgrimaldi) [reply eox_core/api/support/v1/tests/integration/test_views.py:170]: @magajh: Now that you mention it, should we consider using a teardown method to undo all operations made during the tests in the near future? This way, we know running a test suite is an idempotent operation.

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**
